src/utils.py

```diff
@@ -0,0 +1,56 @@
+from typing import Union
+from io import BytesIO
+from PIL import Image
```

Comment:
Like with requirements.txt, mild preference to alphabetize these imports. Also not sure why numpy is separate and not up here: if you've got a good reason, that's great, and if not, alphabetize all 4 of these.